tests/integration/api_v3/http/blocks.test.js

```diff
@@ -0,0 +1,400 @@
+const baseUrlV3 = `${baseUrl}/api/v3`;
+const endpoint = `${baseUrlV3}/blocks`;
+
+const {
+	goodRequestSchema,
+	badRequestSchema,
+	notFoundSchema,
+	wrongInputParamSchema,
+	metaSchema,
+} = require('../../../schemas/httpGenerics.schema');
+
+const {
+	blockSchema,
+} = require('../../../schemas/api_v3/block.schema');
+
+describe('Blocks API', () => {
+	let refBlock;
+	beforeAll(async () => {
+		[refBlock] = (await api.get(`${endpoint}?limit=1&offset=5`)).data;
+	});
+
+	describe('GET /blocks', () => {
+		it('returns list of blocks', async () => {
+			const response = await api.get(`${endpoint}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
+			expect(response.data.length).toBeLessThanOrEqual(10);
+			response.data.forEach((block, i) => {
+				expect(block).toMap(blockSchema);
+				if (i < response.data.length - 1) {
+					expect(block.height).toBe(response.data[i + 1].height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('returns list of blocks when call with limit 10', async () => {
+			const response = await api.get(`${endpoint}?limit=10`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
+			expect(response.data.length).toBeLessThanOrEqual(10);
+			response.data.forEach((block, i) => {
+				expect(block).toMap(blockSchema);
+				if (i < response.data.length - 1) {
+					expect(block.height).toBe(response.data[i + 1].height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('known block by blockID -> ok', async () => {
+			const response = await api.get(`${endpoint}?blockID=${refBlock.id}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data.length).toEqual(1);
+			response.data.forEach((block, i) => {
+				expect(block).toMap(blockSchema, { id: refBlock.id });
+				if (i < response.data.length - 1) {
+					expect(block.height).toBe(response.data[i + 1].height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('known block by height -> ok', async () => {
+			const response = await api.get(`${endpoint}?height=${refBlock.height}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data.length).toEqual(1);
+			expect(response.data[0].height).toEqual(refBlock.height);
+			expect(response.data[0]).toEqual(refBlock);
+			response.data.forEach((block, i) => {
+				expect(block).toMap(blockSchema, { height: refBlock.height });
+				if (i < response.data.length - 1) {
+					expect(block.height).toBe(response.data[i + 1].height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('known block by generatorAddress -> ok', async () => {
+			const response = await api.get(`${endpoint}?generatorAddress=${refBlock.generatorAddress}`);
+			expect(response).toMap(goodRequestSchema);
```

Comment:
Please ensure that the response length is validated.
```suggestion
			expect(response).toMap(goodRequestSchema);
			expect(response.data.length).toBeGreaterThanOrEqual(1);
			expect(response.data.length).toBeLessThanOrEqual(10);
```